system/datatypes/helpPage.php

```diff
@@ -0,0 +1,194 @@
+	 }
+
+	 function userhelp() {
+	global $uiLangList;
+	global $isoLangList;
+	echo ' generaluserhelp';
+			$this->html->writeToElement('main', 'Informatie over de users in Hypha:');
+			ob_start();
+?>
+<div class="help">
+pagasna naam Hier meer uitleg over de help
+tzt vervangen door get-help user uit een xml taal bestand<br>
+</div>
+<?php
+			$this->html->writeToElement('main', ob_get_clean());
+	 }
+	 function infobutton($button) {
+		 // test function for ajax
+		 // returns dummy message to test ajax call
+	 //echo ' infobutton';
+	 //$this->html->writeToElement('main','<span> info for button'. $button. '</span>' );
+	 return 'helpPage button'. $button;
+	}
+
+	function pagenamehelp($pagetype) {
+	global $uiLangList;
+	global $isoLangList;
+	echo 'pagenamehelp';
+			$this->html->writeToElement('main', 'Informatie over een bepaald datatype Pagina type: '.$pagetype);
+			ob_start();
+?>
+<div class="help">
+pagasna naam Hier meer uitleg over de help
+tzt vervangen door get-help user uit een xml taal bestand<br>
+</div>
+<?php
+			$this->html->writeToElement('main', ob_get_clean());
+ }
+
+ function showhelpPageName() {
+ global $uiLangList;
+ global $isoLangList;
+ echo 'generalpagenamehelp';
+		 $this->html->writeToElement('main', 'Hulp bij het maken van een pagina naam');
+		 ob_start();
+?>
+<div class="help">
+pagasna naam Hier meer uitleg over de help
+tzt vervangen door get-help user uit een xml taal bestand<br>
+</div>
+<?php
+		 $this->html->writeToElement('main', ob_get_clean());
+}
+
+function showhelpMenu($menuItem) {
+global $uiLangList;
+global $isoLangList;
+echo 'showHelpmenu';
+		$this->html->writeToElement('main', 'Hulp bij het over een menu item '.$menuItem);
+		ob_start();
+?>
+<div class="help">
+Hoe maak ik een menu
+Hier meer uitleg over de help
+tzt vervangen door get-help user uit een xml taal bestand<br>
+</div>
+<?php
+		$this->html->writeToElement('main', ob_get_clean());
+}
+
+function generalmenuhelp() {
+global $uiLangList;
+global $isoLangList;
+echo 'generalhelpmenu';
+		$this->html->writeToElement('main', 'Hulp bij het aanmaken van een menu');
+		ob_start();
+?>
+<div class="help">
+Hoe maak ik een menu
+Hier meer uitleg over de help
+tzt vervangen door get-help user uit een xml taal bestand<br>
+</div>
+<?php
+		$this->html->writeToElement('main', ob_get_clean());
+}
+
+
+function showhelpindex($language){
+	global $uiLangList;
+	global $isoLangList;
+	//echo 'functie showHelpindex language='.$language;
+  $this->html->writeToElement('main',hypha_helpindex($language));
+}
+
+function gethelpbuttoninfo($button,$_language){
+ return "Help voor" . $button . " in de taal ". $language;
+}
+
+function helpinfo($subject,$language){
+$info =  "HELPPAGES.PHP HELPINFO: argument subject ". $subject . " in de taal ". $language . "<br>\n";
+//	$info = hypha_searchhelp($subject, $language);
+	return $info;
+}
+
+function hypha_searchhelp($subject,$language = 'en') {
+	$subject = str_replace("%20", " ",$subject);
```

Comment:
If this replacement is needed, it should probably happen elsewhere. Urlencoding is associated with parsing and generating URLs, while this code is low-level database-access code, which should not have to know about URLs.